### PR TITLE
feat(site): move proptypes tables to top level of component page

### PR DESCRIPTION
**Pull request by coldpour**
### Description

move proptypes tables to top level of component page

### Motivation and context

the proptables used to be rendered per example, now they're only rendered once per component at the top of the page

connects #53 

### How has this been tested?

ran the demo

### Types of changes

- Other (provide details below)

refactor

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered
* [x] Documentation created or updated

### How does this PR make you feel?

hamburger

### CODE REVIEW

**Comment by coldpour** [comment]
part of #53

**Comment by limscoder** [comment]
We should give priority to getting the button PR in.

**Comment by coldpour** [comment]
That's shouldn't conflict really

**Comment by limscoder** [comment]
lgtm

File: packages/site/src/components/PropTable.js [hunk 1]
```diff
@@ -148,64 +162,39 @@ function PropDoc({
...
 
-export default function PropList({
-  exampleProps = [],
-  onPropChange = () => {}
-}: Props) {
-  if (exampleProps.length < 1) {
-    return null;
-  }
+export default function PropTable({ exampleProps = {} }: Props) {
+  const foo = getExampleProps(exampleProps);
```
Comment (kylegach) [inline packages/site/src/components/PropTable.js]: foo?

File: packages/site/src/components/ComponentDoc.js [hunk 2]
```diff
@@ -184,3 +183,7 @@ function renderExamples(
     </div>
   );
 }
+
+function renderPropDoc(propDoc: Object) {
+  return [<H3 key={0}>Props</H3>, <PropTable key={1} exampleProps={propDoc} />];
```
Comment (kylegach) [inline packages/site/src/components/ComponentDoc.js]: Didn't realize we could return arrays of JSX elements. Will not miss the endless wrapping `div`s.

**Comment by kylegach** [comment]
Please use the PR template as directed. If you'd like to modify the template, we can discuss that in a retro or via another PR with proposed changes.

File: packages/site/src/components/PropTable.js (hunk 1 above)
Comment (brentertz) [inline packages/site/src/components/PropTable.js]: Agreed, foo?

**Comment by brentertz** [comment]
As expressed several times in the past, I would really like to see the type field split out into its own column and not be arbitrarily joined with description.  It seems like a very minor change and at least tangentially associated to this PR.  Can it not be included here?  It would be nice to get this out prior to publishing (hopefully today).

**Comment by coldpour** [comment]
@kylegach, I will update the PR description and follow up with my grievances of the template in a PR to that note

@brentertz, I didn't realize how strongly you felt on this point. Everytime, you brought it up, we'd raise the counterpoint of 3 columns being enough and saving us from empty columns, and then the conversation would stop. I think that should be a separate issue anyway since it is only tangentially related to this one and I'd rather handle separate concerns separately.

File: packages/site/src/components/PropTable.js (hunk 1 above)
Comment (coldpour) [inline packages/site/src/components/PropTable.js]: oooh...shucks

**Comment by brentertz** [comment]
@coldpour No worries about the type column.  We'll tackle it in a followup.

**Pull request merged**